lib/import/colorectal/providers/sheffield/sheffield_handler_colorectal.rb

```diff
@@ -7,618 +7,413 @@ module Providers
-          BOCC = %w[EPCAM MLH1 MSH2 MSH6 PTEN STK11 PMS2].freeze
-          CCP_APCMUTYH = %w[APC MUTYH].freeze
-          CCP_CRC_FULLPANEL = %w[MLH1 MSH2 MSH6 PMS2 EPCAM APC MUTYH
-                                 BMPR1A PTEN POLD1 POLE SMAD4 STK11].freeze
-
-          # Il resto di CCP lo prendo direttamente da karyotyping_method
-          R209_NGS = %w[APC MUTYH].freeze
-          R209_UNK_SMALL = %w[MLH1 MSH2 MSH6 PMS2 EPCAM APC MUTYH BMPR1A
-                              PTEN POLD1 POLE SMAD4 STK11].freeze
-          R209_UNK_MLPA = %w[MLH1 MSH2 APC MUTYH].freeze
-          R210_FULLSCREEN = %w[MLH1 MSH2 MSH6 PMS2 EPCAM].freeze
-          R211_FULLSCREEN = %w[APC MUTYH].freeze
-          COLORECTAL_GENES_REGEX = /(?<colorectal>APC|
-                                                BMPR1A|
-                                                EPCAM|
-                                                MLH1|
-                                                MSH2|
-                                                MSH6|
-                                                MUTYH|
-                                                PMS2|
-                                                POLD1|
-                                                POLE|
-                                                PTEN|
-                                                SMAD4|
-                                                STK11)/xi .freeze # Added by
-          NULL_TARGETED_TEST_REGEX = /(?<colorectal>APC|BMPR1A|EPCAM|
-                                    MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11):
-                                    \sFamilial\spathogenic\smutation\snot\sdetected/ix .freeze
-
-          def initialize(batch)
-            @failed_genotype_counter = 0
-            @successful_gene_counter = 0
-            @gene_counter = 0
-            @failed_gene_counter = 0
-            @negative_test = 0
-            @positive_test = 0
-            super
-          end
+          include Import::Helpers::Colorectal::Providers::Rcu::Constants
 
           def process_fields(record)
+            genotype_str = record.raw_fields['genetictestscope'].strip
+            return if NON_CRC_GENTICTESCOPE.include? genotype_str.downcase
+
             genocolorectal = Import::Colorectal::Core::Genocolorectal.new(record)
             genocolorectal.add_passthrough_fields(record.mapped_fields,
                                                   record.raw_fields,
                                                   PASS_THROUGH_FIELDS_COLO)
-            add_test_scope_from_karyo(genocolorectal, record)
+            add_test_scope_from_geno_karyo(genocolorectal, record)
             add_organisationcode_testresult(genocolorectal)
-            res = add_colorectal_from_raw_test(genocolorectal, record)
+            add_test_type(genocolorectal, record)
+            res = process_variants_from_record(genocolorectal, record)
```

Comment:
Is `res` "results" here? 

I don't think we need to use `res.map` here, the results are not being collected, `each` should be fine

Reply:
agree.. each is needed and not map method here.. corrected now